Userland/Libraries/LibCrypto/PK/RSA.h

```diff
@@ -65,21 +96,41 @@ class RSAPrivateKey {
     Integer const& modulus() const { return m_modulus; }
     Integer const& private_exponent() const { return m_private_exponent; }
     Integer const& public_exponent() const { return m_public_exponent; }
+    Integer const& prime1() const { return m_p; }
+    Integer const& prime2() const { return m_q; }
+    Integer const& exponent1() const { return m_dp; }
+    Integer const& exponent2() const { return m_dq; }
+    Integer const& coefficient() const { return m_qinv; }
     size_t length() const { return m_length; }
-    void set_length(size_t length) { m_length = length; }
 
-    void set(Integer n, Integer d, Integer e)
+    ErrorOr<ByteBuffer> export_as_der() const
     {
-        m_modulus = move(n);
-        m_private_exponent = move(d);
-        m_public_exponent = move(e);
-        m_length = m_modulus.trimmed_length() * sizeof(u32);
+        ASN1::Encoder encoder;
```

Comment:
should this API take an encoder by ref, similar to how our IPC::Encoder works? That way a caller can avoid making a bunch of temp byte buffers, and simply get the byte buffer for the top level item that they want.

Reply:
Nothing embeds this object directly, you'll have to make a temp bytebuffer anyway...?

Reply:
SubjectPublicKeyInfo and PrivateKeyInfo do, no?

Reply:
Not directly, they embed an octet/bit string with the DER-encoded contents of this thing